lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -82,14 +78,32 @@ def list(self, request, course, *args, **kwargs):
             if course_handouts_module.data == "<ol></ol>":
                 handouts_html = None
             else:
-                handouts_html = course_handouts_module.data
-                handouts_html = replace_static_urls(
-                    handouts_html,
-                    course_id=course.id,
-                    static_asset_path=course.static_asset_path
-                )
-                handouts_html = make_static_urls_absolute(self.request, handouts_html)
+                handouts_html = apply_wrappers_to_content(course_handouts_module.data, course_handouts_module, request)
             return Response({'handouts_html': handouts_html})
         else:
             # course_handouts_module could be None if there are no handouts
             return Response({'handouts_html': None})
+
+
+def apply_wrappers_to_content(content, module, request):
+    """
+    Updates a piece of html content with the filter functions stored in its module system, then replaces any
+    static urls with absolute urls.
+
+    Args:
+        content: The html content to which to apply the content wrappers generated for this module system.
+        module: The module containing a reference to the module system which contains functions to apply to the
+        content. These functions include:
+            * Replacing static url's
+            * Replacing course url's
+            * Replacing jump to id url's
+        request: The request, used to replace static URLs with absolute URLs.
+
+    Returns: A piece of html content containing the original content updated by each wrapper.
+
+    """
+    content = module.system.replace_urls(content)
+    content = module.system.replace_course_urls(content)
+    content = module.system.replace_jump_to_id_urls(content)
```

Comment:
I changed from iterating over wrappers to this approach because I think that it is FAR more readable and does not run the risk of unwanted filters being applied to our html content.